content/post/055-js-libp2p-0.26.md

```diff
@@ -0,0 +1,128 @@
+author: Jacob Heun
+---
+
+> Gossipsub support and Promisify API
+
+# 🔦 Highlights
+
+## 🗣 Gossipsub
+Thanks to the awesome work of the [ChainSafe](https://github.com/ChainSafe) team, [Gossipsub](https://github.com/ChainSafe/gossipsub-js) is here! If you are using Pubsub you can now switch to using Gossipsub instead of Floodsub. Have old peers you need to flood stuff to? Don't worry, Gossipsub will automatically fallback to Floodsub for peers that don't support it. See the API Changes section below for how to migrate your config over to using Gossipsub.
+
+## 🚉 Promisify
+As we migrate to async/await we are promisifying the Libp2p public methods. Several lower level libraries are currently leveraging libp2p for testing. Their transition to full async/await will be greatly helped by promisifying the libp2p api. Once the async/await changes are propagated up, we will remove Promisify in a future update, along with callback support, as libp2p will have full async/await support at that time. See the API Changes below to check out the Breaking Changes.
+
+# 🏗 API Changes
+
+## BREAKING CHANGES
+
+### Configuration
+
+#### Switch Options
+
+Configuration for `libp2p-switch` has changed. `blacklistTTL` and `blackListAttempts` have been changed to `denyTTL` and `denyAttempts` respectively. You can set these as follows:
+```js
+const libp2p = new Libp2p({
+  switch: {
+    denyTTL: 120e3,
+    denyAttempts: 5,
+  }
+  ...
+})
+```
+
+#### Pubsub
+
+**Subscribe**: To comply with the pubsub interface, the order of params for `libp2p.pubsub.subscribe` have been reordered. They were previously `topic, options, handler, callback`, and are now `topic, handler, options, callback`. If your implementation does not use `options` you should be able to ignore this change.
+
+**Config**: Pubsub was previously enabled via the `EXPERIMENTAL` config. You must now specify your pubsub implementation (Gossipsub or Floodsub), and enable/disable it via it's own configuration. Setting pubsub will automatically enable it. You can disable it by explicitly setting enabled to false in the configuration .
+
+```js
+const libp2p = new Libp2p({
+  modules: {
+    pubsub: require('libp2p-gossipsub'),
+    ...
+  },
+  config: {
+    pubsub: {
+      enabled: true
+    },
+    ...
+  }
+  ...
+})
+```
+
+## Promisify
+
+All libp2p public methods have been promisified, so callbacks can be omitted in favor of promise based usage, including async/await. For example, `start` can be used as:
+
+```js
+libp2p.start(onStart)
+// or
+await libp2p.start()
+```
+
+## Gossipsub
+
+The Gossipsub API is integrated into the existing `libp2p.pubsub` api. Once Gossipsub has been supplied as your pubsub implementation, you will be able to use it just as Floodsub was previously used. See the section above for pubsub configuration changes.
+
+# ❤️ Huge thank you to everyone that made this release possible
+
+In alphabetical order, here are all the humans that contributed to the release:
```

Comment:
FYI, I tweaked [the script](https://gist.github.com/alanshaw/5a2d9465c5a05b201d949551bdb1fcc3) to output total people and contribution counts like:

```js
`In alphabetical order, here are the ${peoples.length} humans that made ${totalContribs} contributions to this release:\n`
```

I also found a bug in rate limit detection for the `name-your-contributors` module so hopefully that'll be fixed soon https://github.com/mntnr/name-your-contributors/pull/108